lib/src/revset.rs

```diff
@@ -103,10 +103,10 @@ pub enum RevsetEvaluationError {
 }
 
 impl RevsetEvaluationError {
-    pub fn expect_backend_error(self) -> BackendError {
+    pub fn into_backend_error(self) -> BackendError {
         match self {
             Self::StoreError(err) => err,
-            Self::Other(err) => panic!("Unexpected revset error: {err}"),
+            Self::Other(err) => BackendError::Other(err),
```

Comment:
> We may want to create some higher-level `RepoError` to return here,

Can you leave a TODO comment? I rejected this implementation once because `RevsetEvaluationError::Other` shouldn't be a `BackendError` (or `CommitStoreError`.)

Reply:
Done